### PR TITLE
PWA: migrate amp-user-notification to ampdocs

### LINKED ISSUE #10075: AMP Shadow with amp-ad and amp-user-notification
When using amp-ad in combination with amp-user-notification in a PWA (both React and Polymer), ads are not loading.
 
Reproducing the error in Polymer
Sample  [here](https://news-amp-1ec10.firebaseapp.com/article/chrome/3-reasons-why-chromebooks-might-be-good)

 
error
 
Reproducing the error in React
Sample [here](https://news-amp-react.firebaseapp.com/content/1.amp.html), and I got:
react-error

 

 

 
AMP in a PWA without any framework just using AMP Shadow:

It works when type of ad is a9, sample [here](https://news-amp-shadow.firebaseapp.com/articles/1.html).
It works when type of ad is doubleclick, sample [here](https://news-amp-shadow.firebaseapp.com/articles/2.html).

@rudygalfi FYI

### PR DESCRIPTION
**Pull request by dvoytenko**
Closes #10075.

### CODE REVIEW

File: extensions/amp-user-notification/0.1/amp-user-notification.js [hunk 1]
```diff
@@ -475,15 +473,18 @@ export class UserNotificationManager {
...
-      UserNotificationManager);
+export function installUserNotificationManagerForTesting(ampdoc) {
+  registerServiceBuilderForDoc(ampdoc, SERVICE_ID, UserNotificationManager);
 }
 
-installUserNotificationManager(AMP.win);
 
-AMP.registerElement('amp-user-notification', AmpUserNotification, CSS);
+// Register the extension services.
+AMP.extension(TAG, '0.1', function(AMP) {
```
Comment (cvializ) [inline extensions/amp-user-notification/0.1/amp-user-notification.js]: Is there a benefit of making this callback a regular function vs a fat-arrow function?

File: extensions/amp-user-notification/0.1/test/test-amp-user-notification.js [hunk 2]
```diff
@@ -27,14 +30,12 @@ describes.realWin('amp-user-notification', {
     extensions: ['amp-user-notification'],
   },
 }, env => {
-  let sandbox;
```
Comment (cvializ) [inline extensions/amp-user-notification/0.1/test/test-amp-user-notification.js]: Is using the `global.sandbox` a best practice for AMP tests that we should try to use elsewhere?

File: extensions/amp-user-notification/0.1/amp-user-notification.js (hunk 1 above)
Comment (dvoytenko) [inline extensions/amp-user-notification/0.1/amp-user-notification.js]: Hmm. They are all `function()`. Can't quite remember why, I think there was a reason. 

/cc @erwinmombay Do you remember?

File: extensions/amp-user-notification/0.1/test/test-amp-user-notification.js (hunk 2 above)
Comment (dvoytenko) [inline extensions/amp-user-notification/0.1/test/test-amp-user-notification.js]: Yes! Tests should use `global.sandbox` everywhere where describes are used.

**Pull request merged**